src/libs/actions/IOU.js

```diff
@@ -66,14 +66,14 @@ function requestMoney(report, amount, currency, recipientEmail, participant, com
         isNewChat = true;
     }
     let iouReport;
-    const originalIOUStatus = chatReport.hasOutstandingIOU;
-    if (originalIOUStatus) {
+    if (chatReport.iouReportID) {
         iouReport = IOUUtils.updateIOUOwnerAndTotal(
             iouReports[`${ONYXKEYS.COLLECTION.REPORT}${chatReport.iouReportID}`],
             recipientEmail,
             amount,
             currency,
         );
+        iouReport.hasOutstandingIOU = iouReport.total !== 0;
```

Comment:
@amyevans What about updating the `hasOutstandingIOU` in cancelMoneyRequest?

Reply:
Good call. Although at that point, it'd mean all 3 times `IOUUtils.updateIOUOwnerAndTotal()` is called, we have a line following to update `hasOutstandingIOU` - so perhaps I should instead move it into the util function?

Reply:
@amyevans correct, not sure if that method is used anywhere else (might be in the payIOU which is now being reviewed) but if we can incorporate it there easily and nicely it might be better

Reply:
Got it. I searched https://github.com/Expensify/App/pull/12739/files and don't see it in the diff anywhere so I think it's safe

Reply:
Done. Although, any opinion on renaming the function vs keeping it as is?